controllers/machine_controller.go

```diff
@@ -101,6 +102,15 @@ func (r *MachineReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manag
 		return errors.Wrap(err, "failed to add Watch for Clusters to controller manager")
 	}
 
+	// Add index to Machine for listing by Node reference. This indexing is used in both MachineHealthCheck and Machine controllers.
```

Comment:
```suggestion
	// Add index to Machine for listing by Node reference.
```

Reply:
I would prefer to keep this note to avoid any deletion of this as it is being used by MachineHealthCheck too.

Reply:
Suggesting to remove it because we'll have to keep it up to date, and that's probably going to slip in a review